dask/templates/dask-jupyter-ingress.yaml

```diff
@@ -12,35 +12,37 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
+  {{- with .Values.jupyter.ingress.annotations }}
   annotations:
-{{ toYaml .Values.jupyter.ingress.annotations | indent 4 }}
+    {{ . | toYaml | indent 4 }}
+  {{- end }}
 spec:
   {{- with .Values.jupyter.ingress.ingressClassName }}
   ingressClassName: "{{ . }}"
   {{- end }}
-{{- if .Values.jupyter.ingress.tls }}
+  {{- if .Values.jupyter.ingress.tls }}
   tls:
     - hosts:
-      - {{ .Values.jupyter.ingress.hostname | quote }}
+        - {{ .Values.jupyter.ingress.hostname | quote }}
       secretName: {{ .Values.jupyter.ingress.secretName | default (printf "%s-jupyter-tls" (include "dask.fullname" .)) }}
-{{- end }}
+  {{- end }}
   rules:
-  - {{- if ne .Values.jupyter.ingress.hostname "" }}
-    host: {{ .Values.jupyter.ingress.hostname }}
-    {{- end }}
-    http:
-      paths:
-        - path: /
-          {{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}
-          pathType: {{ .Values.jupyter.ingress.pathType }}
-          backend:
-            service:
-              name: {{ template "dask.fullname" . }}-jupyter
-              port:
-                number: {{ .Values.jupyter.servicePort }}
-          {{- else }}
-          backend:
-            serviceName: {{ template "dask.fullname" . }}-jupyter
-            servicePort: {{ .Values.jupyter.servicePort }}
-          {{- end }}
-{{- end -}}
+    - http:
+        paths:
+          - path: /
+            {{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}
+            pathType: {{ .Values.jupyter.ingress.pathType }}
+            backend:
+              service:
+                name: {{ template "dask.fullname" . }}-jupyter
+                port:
+                  number: {{ .Values.jupyter.servicePort }}
+            {{- else }}
+            backend:
+              serviceName: {{ template "dask.fullname" . }}-jupyter
+              servicePort: {{ .Values.jupyter.servicePort }}
+            {{- end }}
+      {{- with .Values.jupyter.ingress.hostname }}
+      host: {{ . }}
+      {{- end }}
```

Comment:
Here I relocated the `host` key to be below the always to be rendered `http` key, and ensured consistent two step indentation for list items under `rules`.